feat(ui): add tap-to-record functionality to custom camera

tsinis opened the pull request:
Hey, @longitachi thanks for a great package!

This PR introduces a new feature to the `ZLCustomCamera` that allows users to record videos with a single tap without holding the record button. This enhancement is designed to improve the user experience, especially when users need to record videos hands-free, such as when self-capturing while dancing, using the selfie stick, tripods, and so on.

## Key Changes

- Added a new boolean property `tapToRecordVideo` to `ZLCameraConfiguration`.
- Introduced a getter `shouldUseTapToRecord` in `ZLCustomCamera` to determine if tap-to-record functionality should be enabled.
- Modified the `setupUI` method in `ZLCustomCamera` to add a tap gesture recognizer for recording when `shouldUseTapToRecord` is `true`.
- Updated the `cameraUsageTipsText` method in `ZLCustomCamera` to return an empty string when `shouldUseTapToRecord` is `true` (it might be improved with an additional "Tap to record video" tip, but sadly I don't speak Chinese and other supported locales).
- Added a new private method `tapToRecordAction` in `ZLCustomCamera` to handle the tap-to-record functionality.
- Enhanced the `startRecord` method in `ZLCustomCamera` to schedule a stop recording action if `shouldScheduleStop` is `true`.
- Updated the `startRecordAnimation` and `stopRecordAnimation` methods in `ZLCustomCamera` to change the color of the recording button to red when `shouldUseTapToRecord` is `true`.

## Benefits

- **User Experience**: This feature makes it more convenient for users to record videos without holding the phone.
- **Consistency**: This behavior aligns with the default behavior of the `UIImagePickerController` video camera, providing a familiar experience for users.

## Non-Breaking Change

This change is non-breaking and does not affect existing functionality. The tap-to-record feature is disabled by default and can be enabled by setting the `tapToRecordVideo` parameter to `true`.

---

Please review the changes and let me know if there are any adjustments or additional features you would like to include. Thank you!

longitachi commented:
Thanks for your pr.
Here is what I would like to change:
* Possibility to move `tapToRecordVideo` to `ZLCameraConfiguration` and prioritize this property lower than `allowTakePhoto`, please note this in the comments. (The default value for `tapToRecordVideo` is false)
* The change to the `ZLCustomeCamera` initialization method could be removed, and then `shouldUseTapToRecord` could be changed to `cameraConfig.tapToRecordVideo && !cameraConfig.allowTakePhoto`.

tsinis commented:
> Thanks for your pr.
> 
> Here is what I would like to change:
> 
> * Possibility to move `tapToRecordVideo` to `ZLCameraConfiguration` and prioritize this property lower than `allowTakePhoto`, please note this in the comments. (The default value for `tapToRecordVideo` is false)
> 
> * The change to the `ZLCustomeCamera` initialization method could be removed, and then `shouldUseTapToRecord` could be changed to `cameraConfig.tapToRecordVideo && !cameraConfig.allowTakePhoto`.

Hey @longitachi, yep this was the only thing I wasn't sure about since this flag is strictly related only to the custom camera. I will do the change in several hours 👍 Thanks!

tsinis commented:
Hey, @longitachi

I have made the requested changes:
- The change to the `ZLCustomCamera` initialization method was removed.
- The `tapToRecordVideo` flag was moved to `ZLCameraConfiguration` and docs are updated.
- `shouldUseTapToRecord` getter using the `tapToRecordVideo` value from the `ZLCameraConfiguration` (see second point).

+ Added the `tapToRecordVideo` method to `ZLCameraConfiguration` to follow the same pattern of assignment, as other flags in this class.
+ Updated comments in the `showCamera()` Example app.

Thanks!

longitachi commented on Sources/Camera/ZLCustomCamera.swift:
```diff
@@ -436,12 +440,17 @@ open class ZLCustomCamera: UIViewController {
+            }
             
             let panGes = UIPanGestureRecognizer(target: self, action: #selector(adjustCameraFocus(_:)))
             panGes.delegate = self
```
Please move these lines of code to the "else" code block above
```swift
if shouldUseTapToRecord {
} else {
    // Move here
    let panGes = UIPanGestureRecognizer(target: self, action: #selector(adjustCameraFocus(_:)))
    panGes.delegate = self
    panGes.maximumNumberOfTouches = 1
    largeCircleView.addGestureRecognizer(panGes)
    cameraFocusPanGes = panGes
}
```
